File: code_samples/go/algorithms_string_algorithms.go
package main

import (
    "fmt"
)

// Algorithms: String Algorithms
// AI/ML Training Sample

type StringAlgorithms struct {
    Data string
}

func NewStringAlgorithms() *StringAlgorithms {
    return &StringAlgorithms{
        Data: "",
    }
}

func (s *StringAlgorithms) Process(input string) {
    s.Data = input
}

func (s *StringAlgorithms) Validate() bool {
    return len(s.Data) > 0
}

func (s *StringAlgorithms) GetData() string {
    return s.Data
}

func main() {
    instance := NewStringAlgorithms()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/lua/algorithms_string_algorithms.lua
-- Algorithms: String Algorithms
-- AI/ML Training Sample

StringAlgorithms = {}
StringAlgorithms.__ index = StringAlgorithms

function StringAlgorithms:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function StringAlgorithms:process(input)
    self.data = input
end

function StringAlgorithms:getData()
    return self.data
end

function StringAlgorithms:validate()
    return #self.data > 0
end

-- Example usage
local instance = StringAlgorithms:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
